models/_sector/nft/trades/chains/ethereum/platforms/sudoswap_v2_ethereum_base_trades.sql

```diff
@@ -0,0 +1,175 @@
+            ,COALESCE(case when cardinality(roy.output_1) = 0 then null else roy.output_1[1] end,cast(0 as uint256)) as royalty_fee_amount_raw
+            ,p.token_contract_address
+            ,p.nft_type
+            ,p.nft_contract_address
+            ,p.pool_type
+            ,p.bonding_curve
+            ,p.pool_address
+            ,sp.call_tx_hash
+            ,sp.call_block_time
+            ,sp.call_block_number
+        FROM (
+            SELECT 
+                *
+                , case when cardinality(call_trace_address) = 0 then array[0] else call_trace_address end as call_trace_address_filled 
+            FROM {{ source('sudoswap_v2_ethereum','LSSVMPair_call_swapTokenForSpecificNFTs') }}
+            WHERE call_success
+            {% if is_incremental() %}
+            AND call_block_time >= date_trunc('day', now() - interval '7' day)
+            {% endif %}
+            ) sp_start
+        LEFT JOIN (
+            SELECT * FROM {{ source('sudoswap_v2_ethereum','ExponentialCurve_call_getBuyInfo') }}
+            UNION ALL 
+            SELECT * FROM {{ source('sudoswap_v2_ethereum','LinearCurve_call_getBuyInfo') }}
+            UNION ALL 
+            SELECT * FROM {{ source('sudoswap_v2_ethereum','XykCurve_call_getBuyInfo') }}
+            UNION ALL 
+            SELECT * FROM {{ source('sudoswap_v2_ethereum','GDACurve_call_getBuyInfo') }}
+        ) sp ON sp_start.call_tx_hash = sp.call_tx_hash
+            AND sp_start.call_block_number = sp.call_block_number
+            AND sp_start.call_trace_address_filled = slice(sp.call_trace_address,1,cardinality(sp_start.call_trace_address_filled))
+            {% if is_incremental() %}
+            AND sp.call_block_time >= date_trunc('day', now() - interval '7' day)
+            {% endif %}
+        LEFT JOIN {{ source('sudoswap_v2_ethereum','RoyaltyEngine_call_getRoyalty') }} roy 
+            ON roy.call_tx_hash = sp.call_tx_hash
+            AND roy.call_block_number = sp.call_block_number
+            AND sp_start.call_trace_address_filled = slice(roy.call_trace_address,1,cardinality(sp_start.call_trace_address_filled))
+            {% if is_incremental() %}
+            AND roy.call_block_time >= date_trunc('day', now() - interval '7' day)
+            {% endif %}
+        LEFT JOIN pools p ON p.pool_address = sp_start.contract_address
+    )
+    
+    , trades as (
+        SELECT
+            'ethereum' as blockchain
+            , 'sudoswap' as project
+            , 'v2' as project_version
+            , call_block_time as block_time
+            , call_block_number as block_number
+            , call_tx_hash as tx_hash
+            , pool_address as project_contract_address
+            , case when trade_category = 'sell' then pool_address else trade_recipient end as buyer
+            , case when trade_category = 'buy' then pool_address else trade_recipient end as seller
+            , nft_contract_address
+            , one_nft_token_id as nft_token_id
```

Comment:
For ERC1155 pools, there's no `token_id` passed into the calls, we should source this token id from the pool creation. 
On top of that, for ERC1155 pools the `nftIds` field on the swap calls is filled with the `numNFTs` data. 
to quote the sudoswap devs: 
["// @dev this is a bit hacky, to allow for better interop w/ other pair interfaces"](https://github.com/sudoswap/lssvm2/blob/4b0390f12ec1b05f69f407471e95eb713cff0f49/src/erc1155/LSSVMPairERC1155.sol#L101)

take a look at this example tx: `0xa253119c7748456c7d383fcbdd03c0cf4897b5d59c697443932bfc18aab855a0`

Reply:
wow, good find! fixing